### PR TITLE
[stable] Microsoft.Windows.Storage.Pickers APIs

### PR DESCRIPTION
**Pull request by DinahK-2SO**
## Description

This PR implements suggestions raised in the official API review of `Microsoft.Windows.Storage.Pickers` APIs (#5634 ).

This pull request introduces significant changes to the `StoragePickers` implementation, primarily focusing on removing the `SettingsIdentifier` property, refining file path validation, and updating related tests and configurations. These changes simplify the API, improve validation logic, and ensure better handling of edge cases.

### 2. Replace method `TrySetSuggestedSaveFilePath` with the property's own setter:
* Added a new method `ValidateSuggestedSaveFilePath` in `PickerCommon.cpp` to validate the syntax and structure of suggested save file paths. The validation method the verifies that

    1. the suggested file's folder is specified, 
    2. the folder's specified with syntactic correctness, 
    3. the validation method does not check folder existence. ( [dev/Interop/StoragePickers/PickerCommon.cppR232-R266](diffhunk://#diff-e7b7de95380ad15538bb5df8e17b272a50b5b2c2a203cef517e803a1d5665823R232-R266) )

* Remove the `TrySetSuggestedSaveFilePath` method in `FileSavePicker.cpp` to replace it with a more robust `SuggestedSaveFilePath` setter that uses the new validation method. ([[1]](diffhunk://#diff-59e025f2c0bb98d7a12bb6b5d5bdcc1f63b0ee4f150ef4fde1338a5de4650fc4L73-R67), [[2]](diffhunk://#diff-31e8543f64c4b9c27fdc095f3d4ac245d2c03be85eed061919cb6cd6c3de3ca1L33-R30))

### 3. For FileSavePicker, updated underlying logic to ensure that the file name in `SuggestedSaveFilePath` takes precedence over the `SuggestedFileName`, even if the folder of `SuggestedSaveFilePath` does not exists.

This is also the existing behavior of UWP pickers ( the `Windows.Storage.Pickers`)

### 4. Test Additions and Updates:
* Added/Updated unit tests in `PickerCommonTests.cpp` to verify the behavior of the `SuggestedSaveFilePath` property when the parent folder is missing or invalid. These tests ensure fallback mechanisms function as expected. 
    * (new) VerifyConfigureFileSaveDialog_WhenSuggestedSaveFilePathFolderDeleted_ExpectItsFileNameStillWork
    * (new) VerifyConfigureFileSaveDialog_WhenSuggestedSaveFilePathFileNameEmptyAndFolderDeleted_ExpectItsFileNameStillWork
    * (updated logic) VerifyFilters_FileOpenPickerWhenFileTypeFiltersDefinedExpectAddingUnionedType
    * (cleaned unused object) VerifyConfigureDialog_WhenPickerParameters_FileTypeFilterNotSpecified_ExpectSuccess
[test/StoragePickersTests/PickerCommonTests.cppR82-R127](diffhunk://#diff-c0106cdb512c855fd3f2a703625b8ea80fe7faf9dc3db1d74149691ef7b945c5R82-R127) 
* Removed obsolete test setup related to `SettingsIdentifier`. ([test/StoragePickersTests/PickerCommonTests.cppL71-L73](diffhunk://#diff-c0106cdb512c855fd3f2a703625b8ea80fe7faf9dc3db1d74149691ef7b945c5L71-L73))

### 5. XML and Feature Flag Cleanup:
* Removed the `alwaysDisabledChannelTokens` section from `TerminalVelocityFeatures-StoragePickers.xml`, indicating that the feature is now always enabled. (`[dev/Common/TerminalVelocityFeatures-StoragePickers.xmlL15-L18](diffhunk://#diff-a948a97cebc67c2afb5f0d64c945496bfdf81c30c3a4b8c7991dc8242449e250L15-L18)`)
* Removed feature flag annotations from `Microsoft.Windows.Storage.Pickers.idl`. ([[1]](diffhunk://#diff-56ef61cce4d5ac680858146bab3f61a75a8e5aa32553a0aac1191cb62ead3d20L4-L19), [[2]](diffhunk://#diff-56ef61cce4d5ac680858146bab3f61a75a8e5aa32553a0aac1191cb62ead3d20L34-L47))

### CODE REVIEW

File: dev/Interop/StoragePickers/PickerCommon.cpp [hunk 1]
```diff
@@ -234,6 +229,41 @@ namespace PickerCommon {
...
+            // allow empty path.
+            return;
+        }
+
+        ValidateStringNoEmbeddedNulls(path);
+
+        std::filesystem::path p;
+        try {
+            // validate that the path string is syntactically correct, does not validate its existence.
+            p = std::filesystem::path(path.c_str());
```
Comment (codendone) [inline dev/Interop/StoragePickers/PickerCommon.cpp]: Does std::filesystem::path actually validate the syntax? In a quick test, I was able to set the path to `"C:\\>:di*r\\file::name.txt"`, so it seems it doesn't validate?

**Review by codendone — APPROVED** [review]
My one comment could be investigated/handled for a follow-up change. The rest of this PR looks good.

**Review by Hong-Xiang — APPROVED** [review]

File: dev/Interop/StoragePickers/PickerCommon.cpp (hunk 1 above)
Comment (DinahK-2SO) [reply dev/Interop/StoragePickers/PickerCommon.cpp]: Thanks @codendone for catching this error! I updated the validation logic to use `SHSimpleIDListFromPath`, as Raymon recommended during the review.
